hibernate-core/src/main/java/org/hibernate/engine/spi/SessionFactoryImplementor.java

```diff
@@ -315,4 +317,8 @@ default <T> RootGraphImplementor<T> createEntityGraph(Class<T> entityType) {
 	 */
 	String bestGuessEntityName(Object object);
 
+	default JdbcSelectWithActionsBuilder getJdbcSelectWithActionsBuilder(){
```

Comment:
```suggestion
    @Incubating
	default JdbcSelectWithActionsBuilder getJdbcSelectWithActionsBuilder(){
```

I wonder if it might be better to pass some of these values directly to this method.  I'd prefer to make it incubating so we can make that change later if we want.